torch/fx/experimental/symbolic_shapes.py

```diff
@@ -916,9 +942,258 @@ def _print_Symbol(self, expr) -> str:
+        #     a) Generate the actual guard expression
+        guard_expression = " && ".join(exprs)
+        #     b) Generate the extra temporary variables corresponding to Tensor
+        #        properties (e.g. sizes, strides, and storage_offset)
+        source_creation = []
+        for i, info in enumerate(self.parameter_infos):
+            if info.type == SymInt:
+                source_creation.append(f"int64_t {info.name} = THPUtils_unpackLong(inputs[{i}]);")
+
+            elif info.type == torch.Tensor:
+                tensor_name = self.register_unique_name(f"{info.name}_tensor")
+                source_creation.append(f"const auto& {tensor_name} = THPVariable_Unpack(inputs[{i}]);")
+
+                for tensor_property in TensorProperty:
+                    if tensor_property == TensorProperty.STORAGE_OFFSET:
+                        tensor_property_str = "storage_offset"
+                        tensor_property_source = TensorPropertySource(info.source, tensor_property)
+                    else:
+                        tensor_property_str = f"{tensor_property.name.lower()}s"
+                        tensor_property_source = TensorPropertySource(info.source, tensor_property, idx=0)
+
+                    if not self.is_source_registered(tensor_property_source):
+                        # Skip if there are no SymInts tracked for this property.
+                        # Usually true for scalar tensors.
+                        continue
+
+                    tensor_property_info = self.get_source_info(tensor_property_source)
+                    source_creation.append(
+                        f"auto {tensor_property_info.name} = {tensor_name}.{tensor_property_str}();"
+                    )
+
+        source_creation_str = "\n".join(" " * 4 + line for line in source_creation)
+        #     c) Generate the C++ source corresponding to this guard.
+        cpp_code = f"""\
+#include <torch/python.h>
+#include <torch/csrc/utils/python_numbers.h>
+#include <cmath>
+extern "C" int guard(PyObject** inputs) {{
+{source_creation_str}
+    return {guard_expression};
+}}
+"""
+
+        #     d) Compile the generated C++ source and cache it
+        try:
+            from torch._inductor.codecache import CppCodeCache
+            with no_conda_cpp_compiler():
+                fn = CppCodeCache.load(cpp_code, include_pytorch=True).guard
+        except Exception:
+            print(f"Code: \n{cpp_code}")
+            raise
+
+        # Before actually executing the compiled function, we need to
+        # preprocess the tensor variables to what numba is actually expecting
+        # (according to the signature we gave above).
+        # In other words, we need to get a pointer to the actual tensor. In
+        # cpython, that can be done by calling the id function.
+        def wrapper_fn(*args):
+            args = (ctypes.c_void_p * len(args))(*[id(a) for a in args])
+            return bool(fn(args))
```

Comment:
Can you not use [ctypes.py_object](https://docs.python.org/3/library/ctypes.html#ctypes.py_object) here?

Reply:
Good question. I will try it out.